### ISSUE TITLE
Setting runtime fraction of numerator 0 and FractionalPercent_HUNDRED for RequestMirrorPolicy ends up using FractionalPercent_THOUSAND

### ISSUE THREAD

**Issue by lita** [envoyproxy/envoy#8299]
**WARNING: If you want to report crashes, leaking of sensitive information,
and/or other security issues, please consider
[reporting them using appropriate channels](https://github.com/envoyproxy/envoy#reporting-security-vulnerabilities).**

**Issue Template**

*Title*: *Setting runtime fraction of numerator 0 and FractionalPercent_HUNDRED for RequestMirrorPolicy ends up using FractionalPercent_THOUSAND*

*Description*:
We are rolling our RequestMirrorPolicy, we set the path within the Runtime Fraction but not the default like so:

```go
RuntimeFraction: &core.RuntimeFractionalPercent{
    RuntimeKey: policy.RuntimeKey,
    DefaultValue: &envoytype.FractionalPercent{
      Numerator:   0,
      Denominator: envoytype.FractionalPercent_HUNDRED,
    },
},
```

But as I change the runtime flag, I am noticing that the runtime fraction is using `FractionalPercent_THOUSAND` as a denominator based on the aggregated stats.

Since `envoytype.FractionalPercent_HUNDRED` is the default value, I believe envoy is treating `DefaultValue` as a nil value and exercising this code path: https://github.com/envoyproxy/data-plane-api/blob/master/envoy/api/v2/route/route.proto#L640-L645

**Comment by zuercher** [comment]
The bug is here: https://github.com/envoyproxy/envoy/blob/master/source/common/router/config_impl.cc#L182

The code does not distinguish between a) the deprecated RequestMirrorPolicy.runtime_key field being set with no RequestMirrorPolicy.runtime_fraction and b) RequestMirrorPolicy.runtime_fraction where numerator = 0.

**Comment by lita** [comment]
Yeah, there is no rush on Lyft's end, as I have change the control plane to use TEN_THOUSAND as a workaround. But wanted to document it. This would be a good starter task for a beginner to do.

**Comment by zuercher** [comment]
The other half of the breadcrumb is https://github.com/envoyproxy/envoy/blob/master/source/common/router/router.cc#108
